packages/node_modules/pouchdb-abstract-mapreduce/src/index.js

```diff
@@ -378,14 +381,14 @@ function createAbstractMapReduce(localDocName, mapper, reducer, ddocValidator) {
   // returns a promise for a list of docs to update, based on the input docId.
   // the order doesn't matter, because post-3.2.0, bulkDocs
   // is an atomic operation in all three adapters.
-  function getDocsToPersist(docId, view, docIdsToChangesAndEmits) {
-    var metaDocId = '_local/doc_' + docId;
-    var defaultMetaDoc = {_id: metaDocId, keys: []};
-    var docData = docIdsToChangesAndEmits.get(docId);
-    var indexableKeysToKeyValues = docData[0];
-    var changes = docData[1];
-
-    function getMetaDoc() {
+  async function getDocsToPersist(docId, view, docIdsToChangesAndEmits) {
+    const metaDocId = '_local/doc_' + docId;
+    const defaultMetaDoc = {_id: metaDocId, keys: []};
+    const docData = docIdsToChangesAndEmits.get(docId);
+    const indexableKeysToKeyValues = docData[0];
+    const changes = docData[1];
+
+    async function getMetaDoc() {
```

Comment:
This function doesn't strictly need to be marked `async`; it doesn't use the `await` keyword and it returns a `Promise` explicitly. It doesn't do any harm to leave this keyword here though.

Reply:
This was removed here https://github.com/pouchdb/pouchdb/pull/8489/commits/cd47d29ff90506541b5bf030998dca635a91d720